Framework/AODMerger/src/aodThinner.cxx

```diff
@@ -149,27 +198,60 @@ int main(int argc, char* argv[])
       exitCode = 7;
       break;
     }
-    auto v0s = (TTree*)inputFile->Get(Form("%s/%s", dfName, "O2v0_001"));
+    auto v0s = (TTree*)inputFile->Get(Form("%s/%s", dfName, v0Name.Data()));
     if (v0s == nullptr) {
-      printf("O2v0_001 table not found\n");
+      printf("%s table not found\n", v0Name.Data());
       exitCode = 8;
       break;
     }
 
+    std::unordered_map<int, bool> keepV0TPCs;
+    if (bV0TPC) {
+      // We need to loop over the V0s once and flag the prong indices
+      // technically we need to flag only V0s with TPC-only prongs but
+      // this requires in-depth knowledge of the data-model which we don't need.
+      int trackIdxPos = 0, trackIdxNeg = 0;
+      v0s->SetBranchAddress("fIndexTracks_Pos", &trackIdxPos);
+      v0s->SetBranchAddress("fIndexTracks_Neg", &trackIdxNeg);
+      for (int i{0}; v0s->LoadTree(i) >= 0; ++i) {
```

Comment:
Why not `i < v0s->GetEntriesFast()` ?

Reply:
honestly, I do not know. I have to look up every time how to iterate a tree and this changed some time ago: https://root.cern/manual/trees/#reading-a-tree. If preferred I can change it.

Reply:
I think there are several ways but if you do `LoadTree` and `GetEntry` you are doing twice the same thing. We use also `GetEntriesFast` in other places in the same file, so I prefer that.